instance/instance.go

```diff
@@ -32,9 +33,13 @@ const (
 
 //GetList a list of all instances
 func (s *Store) GetList(w http.ResponseWriter, r *http.Request) {
-	stateFilter := r.URL.Query().Get("state")
+	stateFilterQuery := r.URL.Query().Get("state")
+	var stateFilterList []string
+	if stateFilterQuery != "" {
+		stateFilterList = strings.Split(stateFilterQuery, ",")
+	}
```

Comment:
Probably should validate the query list contains only valid values before going any further. If not you can return an appropriate HTTP status code at this point rather an running a query which does not return any results.

Reply:
Also probably want to trim any whitespace, lower or upper case everything.

Reply:
I agree with Dave on both cases, but you will want to extend unit tests to cover for these scenarios as well as extending the API integration tests that I mentioned on my review.

Reply:
That's helpful thanks. By 'valid values' do we mean a value that the 'state' property can be set to? Because I can set that state property to any string at the moment.

Reply:
Yeah - so a state can be one of a finite set of values - we should check that what we have is one of those before hitting the DB. The additional benefit is that if we receive junk we can send back a meaningful HTTP response code - something that we might not be able to do further into the app.

Reply:
I'm just putting this here as a record for future us:
We've had a chat and agreed to record this as technical debt because it's not an issue specific to this application and we should probably be validating what gets put in the database, not just that the queries match.